modules/core/include/opencv2/core/types.hpp

```diff
@@ -557,7 +558,9 @@ class CV_EXPORTS RotatedRect
     Point2f center;
     //! returns width and height of the rectangle
     Size2f size;
-    //! returns the rotation angle. When the angle is 0, 90, 180, 270 etc., the rectangle becomes an up-right rectangle.
+    /** returns the rotation angle, based on the top edge(topLeft-topRight) of the rectangle.
```

Comment:
Rotation is not based on any edge. Rotation has direction (clockwise here), center (_center_ field) and an angle (this field). This addition makes documentation more confusing.

Reply:
`based` means where the 0-degree starts from
```suggestion
    /** returns the rotation angle, that is the angle between the perpendicular from the center point to the top edge(topLeft-topRight) of the rotated rectangle, compares to the up-right rectangle.
```

Reply:
There is no 0-degree vector in this case. It just does not matter.

Reply:
https://github.com/opencv/opencv/issues/23335#issuecomment-1463706317
> we can say that at 0 degrees, the top edge (1-2) is at the top and the rectangle "is pointing north".

I just want to mention anyone who may get confused by the pointing direction of the angle.

Reply:
Angle does not have the pointing direction, it is a scalar.